File: minjung/프로그래머스/완전탐색/카펫.py
def solution(brown, yellow):
    # 전체 면적 = brown+yellow의 갯수 = height*width
    s = brown +yellow
    ans = [0,0]
    
    for width in range(s-1,0,-1):
        if s%width :
            continue
        height = s/width;
        y = (width-2)*(height-2)
        b = s-y
        if(y==yellow and b==brown):
            ans[0] = width
            ans[1] = height
            break
    
    return ans

File: chaeeun/Programmers/Lv2/카펫.js
function solution(brown, yellow) {
    var answer = [];
    
    // 약수들로 접근!
    var arr = []
    for (let i = 1; i <= yellow; i++) {
        if (yellow % i == 0) {
            if (i <= yellow/i) {
                if ((i*2)+((yellow/i)*2)+4 == brown) {
                    answer = [yellow/i+2, i+2]
                }
            } else {
                break
            }
        }
    }
    
    return answer;
}